### PR TITLE
Refactored full category overview + translate function

### PR DESCRIPTION
**Pull request by modelrailroader**
Refactored both pages
Moved to twig
Fixed minor bugs with translating categories

### CODE REVIEW

File: phpmyfaq/src/phpMyFAQ/Category.php [hunk 1]
```diff
@@ -103,9 +103,10 @@ class Category
      */
     public function __construct(
         private readonly Configuration $configuration,
-        array $groups = [],
-        bool $withPerm = true
-    ) {
+        array                          $groups = [],
```
Comment (thorsten) [inline phpmyfaq/src/phpMyFAQ/Category.php]: Please only one space between the type and the variable name

File: phpmyfaq/src/phpMyFAQ/Category.php [hunk 2]
```diff
@@ -718,17 +719,18 @@ public function getCategoryData(int $categoryId): CategoryEntity
...
-     * @param int    $id Entity ID
+     * @param int $id Entity ID
      * @param string $separator Path separator
-     * @param bool   $renderAsHtml Renders breadcrumbs as HTML
+     * @param bool $renderAsHtml Renders breadcrumbs as HTML
      * @param string $useCssClass Use CSS class "breadcrumb"
      */
     public function getPath(
-        int $id,
+        int    $id,
```
Comment (thorsten) [inline phpmyfaq/src/phpMyFAQ/Category.php]: same here

File: phpmyfaq/assets/templates/admin/content/category.showstructure.twig [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  {% for category in categoryTree %}
+    <tr>
+      {% set indent = category.indent * 2 %}
+      <th>
+        <a href="?action=editcategory&cat={{ category.id }}"><span><i class="bi bi-pencil"></i></span></a>&nbsp;
+        <span class="ps-{{ indent }}">{{ category.name }}</span>
+      </th>
+      {% for code in all_lang_codes %}
+        <th class="text-center">
+          {% for catid, category_trans in translations %}
```
Comment (thorsten) [inline phpmyfaq/assets/templates/admin/content/category.showstructure.twig:49]: We should use camelCase instead of undescores in variable names

File: phpunit.xml [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
-  <source>
-    <include>
-      <directory suffix=".php">./phpmyfaq/src/phpMyFAQ</directory>
-    </include>
-    <exclude>
-      <directory suffix=".php">./phpmyfaq/src/libs</directory>
-    </exclude>
-  </source>
-</phpunit>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (thorsten) [inline phpunit.xml:1]: Please revert this change

**Review by thorsten — CHANGES_REQUESTED** [review]

File: phpmyfaq/src/phpMyFAQ/Category.php (hunk 1 above)
Comment (modelrailroader) [reply phpmyfaq/src/phpMyFAQ/Category.php]: This was part of my PHPStorm-Problem but it is outdated.

File: phpmyfaq/src/phpMyFAQ/Category.php (hunk 2 above)
Comment (modelrailroader) [reply phpmyfaq/src/phpMyFAQ/Category.php]: Same here, outdated.

**Comment by modelrailroader** [comment]
Two change requests were completed, two are outdated. PR is ready to merge @thorsten
